packages/router/router.ts

```diff
@@ -2899,6 +2914,9 @@ function processRouteLoaderData(
         errors[boundaryMatch.route.id] = error;
       }
 
+      // Clear our any prior loaderData for the throwing route
+      loaderData[id] = undefined;
```

Comment:
If this loader threw an error - we mark this to ensure that we don't hang onto it's prior `loaderData` during the merge